app/services/lifetime_usages/flag_refresh_from_invoice_service.rb

```diff
@@ -31,8 +31,8 @@ def call
 
     attr_reader :invoice
 
-    def has_plan_usage_thresholds?
-      invoice.subscriptions.any? { |s| s.plan.usage_thresholds.any? }
+    def should_flag_refresh_from_invoice?
+      invoice.organization.lifetime_usage_enabled? || invoice.subscriptions.any? { |s| s.plan.usage_thresholds.any? }
```

Comment:
It was already like this and it's not the purpose of this PR so we can discuss it elsewhere but I think there is a double N+1 so how do you feat about this:

```diff
- invoice.subscriptions.any? { |s| s.plan.usage_thresholds.any? }
+ UsageThreshold.where(plan_id: invoice.subscriptions.map(&:plan_id).any?
```

Reply:
I don't disagree, but we're likely only going to have a couple of subscriptions per invoice, I don't think it's worth it to optimize this one just yet. (given that the new version is not as readable)